pkg/network/config/sysctl/sysctl_test.go

```diff
@@ -0,0 +1,102 @@
+// +build linux
+
+package sysctl
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestString(t *testing.T) {
+	// bogus sysctl
+	_, err := NewString("/tmp", "foo", 0).Get()
+	require.Error(t, err)
+
+	s, err := NewString("/proc", "net/ipv4/tcp_keepalive_intvl", 0).Get()
+	require.NoError(t, err)
+	require.NotEmpty(t, s)
+
+	procRoot := createTmpProcSys(t)
+	defer os.RemoveAll(procRoot)
+
+	createTmpSysctl(t, procRoot, "foo", "bar\n")
+	sctl := NewString(procRoot, "foo", 10*time.Second)
+	v, err := sctl.Get()
+	require.NoError(t, err)
+	require.Equal(t, "bar", v)
+	// update the value on disk
+	createTmpSysctl(t, procRoot, "foo", "baz\n")
+	// check if value is still cached
+	v, err = sctl.Get()
+	require.NoError(t, err)
+	require.Equal(t, "bar", v)
+
+	createTmpSysctl(t, procRoot, "foo", "bar\n")
+	// sysctl with a shorter ttl
+	sctl = NewString(procRoot, "foo", 2*time.Second)
+	v, err = sctl.Get()
+	require.NoError(t, err)
+	require.Equal(t, "bar", v)
+
+	createTmpSysctl(t, procRoot, "foo", "baz")
+	require.Eventually(t, func() bool {
+		v, _ := sctl.Get()
+		return v == "baz"
+	}, 3*time.Second, 500*time.Microsecond)
```

Comment:
For this kind of test, I prefer to to create a helper which takes in time.Now() and test the helper. 

```
func Get() foo {
 return getWithTime(time.Now())
}

func getWithTime(now time.Time) foo
}
```

This makes the test run faster. Only a 3 second speedup for this test, but slow tests add up.  It also makes the test less sensitive to the whims to the go scheduler.